### PR TITLE
[fix][cli] Fix the pulsar-daemon parameter passthrough syntax

### PR DESCRIPTION
**Pull request by coderzc**
### Motivation

Using `bin/pulsar-daemon start standalone` can't start pulsar due to https://github.com/apache/pulsar/pull/22867 changing `bin/pulsar` parameter passthrough syntax.

### Modifications

replace `"$1"` with `"$@"` in bin/pulsar

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: bin/pulsar-daemon [hunk 1]
```diff
@@ -157,7 +157,7 @@ start ()
     echo starting $command, logging to $logfile
     echo Note: Set immediateFlush to true in conf/log4j2.yaml will guarantee the logging event is flushing to disk immediately. The default behavior is switched off due to performance considerations.
     pulsar=$PULSAR_HOME/bin/pulsar
-    nohup $pulsar $command "$1" > "$out" 2>&1 < /dev/null &
+    nohup $pulsar $command $1 > "$out" 2>&1 < /dev/null &
```
Comment (lhotari) [inline bin/pulsar-daemon:160]:
```suggestion
    nohup $pulsar $command "$@" > "$out" 2>&1 < /dev/null &
```

Comment (lhotari) [reply bin/pulsar-daemon:160]: using `"$@"` is the correct thing to do.

Another change is also needed. `start "$*"` should be replaced with `start "$@"` on line 219.

**Review by lhotari — CHANGES_REQUESTED** [review]

File: bin/pulsar-daemon (hunk 1 above)
Comment (lhotari) [reply bin/pulsar-daemon:160]: the solution for restart needs also fixing. I can push changes directly to this PR to help in fixing that.

Comment (coderzc) [reply bin/pulsar-daemon:160]: In my local machine, it cannot be started using quotation marks. I need to remove the quotation to make it work.

**Review by lhotari — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

**Review by shibd — APPROVED** [review]

**Pull request merged**